dkulib/dku_io_utils/column_descriptions.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+"""Module with read/write utility to set dataset column descriptions based on the Dataiku API"""
+
+from typing import Dict
+
+import dataiku
+
+
+def set_column_descriptions(
+    output_dataset: dataiku.Dataset, column_descriptions: Dict, input_dataset: dataiku.Dataset = None
+) -> None:
+    """Set column descriptions of the output dataset based on a dictionary of column descriptions
+
+    Retain the column descriptions from the input dataset if the column name matches.
+
+    Args:
+        output_dataset: Output dataiku.Dataset instance
+        column_descriptions: Dictionary holding column descriptions (value) by column name (key)
+        input_dataset: Optional input dataiku.Dataset instance
+            in case you want to retain input column descriptions
+
+    """
+    output_dataset_schema = output_dataset.read_schema()
+    input_dataset_schema = []
+    input_columns_names = []
+    if input_dataset is not None:
+        input_dataset_schema = input_dataset.read_schema()
+        input_columns_names = [col["name"] for col in input_dataset_schema]
+    for output_col_info in output_dataset_schema:
+        output_col_name = output_col_info.get("name", "")
+        output_col_info["comment"] = column_descriptions.get(output_col_name)
+        if output_col_name in input_columns_names:
+            matched_comment = [
+                input_col_info.get("comment", "")
```

Comment:
I think it would be more intuitive to first set all column descriptions to those of the input & then update the ones which have changed in the output rather than the other way round

In the current implementation if I feed in an output dataset with a new description that was empty previously and I also feed in an input dataset because I want to keep some descriptions, it replaces all my new descriptions with "" (Perhaps I'm doing something wrong)

Reply:
Interesting, that seems unexpected. Can you detail what you had as input and what you received as output?

Reply:
Sure here's the input:

Output dataset; See schema:
[{'name': 'order_date', 'type': 'string'}, {'name': 'pages_visited', 'type': 'bigint'}, {'name': 'order_id', 'type': 'string'}, {'name': 'customer_id', 'type': 'string'}, {'name': 'tshirt_category', 'type': 'string'}, {'name': 'tshirt_price', 'type': 'double'}, {'name': 'tshirt_quantity', 'type': 'bigint'}] 

Column descriptions:
{'order_date': 'Foo'} 

Input dataset; See schema:
[{'name': 'order_date', 'type': 'string', 'comment': ''}, {'name': 'pages_visited', 'type': 'string'}, {'name': 'order_id', 'type': 'string'}, {'name': 'customer_id', 'type': 'string'}, {'name': 'tshirt_category', 'type': 'string'}, {'name': 'tshirt_price', 'type': 'string'}, {'name': 'tshirt_quantity', 'type': 'string'}]

Because the comment is an empty string for order_date it doesnt add the new description. The empty string results from me applying it a couple times I guess

Reply:
Very good catch! Let me rewrite the function as you advised. Should be simpler and fix the case you described.

Reply:
Hi Niklas, I am not able to reproduce the issue. Do you have an example project export or a screen recording to explain what you mean? Thanks!

Reply:
Thanks a lot Niklas. I have been able to reproduce thanks to the project export you shared on Slack. Here is a refactored version according to your feedback: 46c9d02
Let me know what you think.